pkg/queue/manager.go

```diff
@@ -808,11 +808,23 @@ func (m *Manager) queueSecondPass(ctx context.Context, w *kueue.Workload) {
 	}
 }
 
-func (m *Manager) HeapifyAllClusterQueues() {
+func (m *Manager) HeapifyClusterQueuesWithEntryPenalties() {
 	m.Lock()
 	defer m.Unlock()
-	for _, cq := range m.hm.ClusterQueues() {
-		if cq != nil {
+
+	if !m.afsEntryPenalties.HasAny() {
+		return
+	}
+
+	clusterQueuesWithPenalties := make(map[kueue.ClusterQueueReference]bool)
```

Comment:
use sets.Set[kueue.ClusterQueueReference] instead, to better reflect the intention